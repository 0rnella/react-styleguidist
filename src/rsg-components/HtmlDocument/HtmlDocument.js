import React, { PropTypes } from 'react';
import HTMLDocument from './HtmlDocumentRenderer';

function getScrtips(assets) {
	return Object
		.keys(assets)
		.map(key => {
			return assets[key].indexOf('server') === -1 ? assets[key] : null;
		})
		.filter(Boolean);
}

function HtmlDocument({ assets = {}, children, title }) {
	// TODO: Handle CSS and JS assets separately
	return (
		<HTMLDocument
			title={ title }
			scripts={getScrtips(assets)}
			stylesheets={assets.stylesheets}
		>
			{ children }
		</HTMLDocument>
	);
}

HtmlDocument.propTypes = {
	assets: PropTypes.object,
	children: PropTypes.node,
	title: PropTypes.string.isRequired,
};

export default HtmlDocument;
