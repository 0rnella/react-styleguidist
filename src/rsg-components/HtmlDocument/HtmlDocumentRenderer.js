import React, { PropTypes } from 'react';
import HTMLDocument from 'react-html-document';

function HtmlDocumentRenderer({ children, scripts, stylesheets, ...props }) {
	return (
		<HTMLDocument
			{...props}
			scripts={scripts}
			stylesheets={stylesheets}
		>
			{ children }
		</HTMLDocument>
	);
}

HtmlDocumentRenderer.propTypes = {
	scripts: PropTypes.array,
	stylesheets: PropTypes.array,
	children: PropTypes.node,
	title: PropTypes.string.isRequired,
};

export default HtmlDocumentRenderer;
