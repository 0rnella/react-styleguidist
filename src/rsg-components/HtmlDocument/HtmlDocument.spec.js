import React from 'react';
import HtmlDocument from './HtmlDocument';
import Renderer from './HtmlDocumentRenderer';

describe('HtmlDocument', () => {
	it('should include assets', () => {
		const actual = shallow(
			<HtmlDocument
				title="Test document"
				assets={{ main: '/main.js' }}
			>
				<span>Hello</span>
			</HtmlDocument>
		);

		expect(actual.prop('scripts')).toContain('/main.js');
	});

	it('should not include server assets', () => {
		const actual = shallow(
			<HtmlDocument
				title="Test document"
				assets={{ main: 'build/main.js', server: 'build/server.js' }}
			>
				<span>Hello</span>
			</HtmlDocument>
		);

		expect(actual.prop('scripts')).not.toContain('server');
	});
});

describe('HtmlDocumentRenderer', () => {
	it('should render children', () => {
		const actual = render(
			<Renderer title="Test document">
				<span>Hello</span>
			</Renderer>
		);

		expect(actual).toMatchSnapshot();
	});
});
