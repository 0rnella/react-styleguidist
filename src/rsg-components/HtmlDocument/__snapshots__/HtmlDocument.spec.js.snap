exports[`HtmlDocumentRenderer should render children 1`] = `
<html>
  <head>
    <title>
      Test document
    </title>
  </head>
  <body>
    <div
      id="app">
      <span>
        Hello
      </span>
    </div>
  </body>
</html>
`;
